contrib/pmsnare/pmsnare.c

```diff
@@ -81,94 +84,93 @@ CODESTARTisCompatibleWithFeature
 		iRet = RS_RET_OK;
 ENDisCompatibleWithFeature
 
-
 BEGINparse
 	uchar *p2parse;
 	int lenMsg;
-	int snaremessage;
-	int tablength;
-
+	int snaremessage; /* 0 means not a snare message, otherwise it's the index of the tab after the tag  */
+	
 CODESTARTparse
-	#define TabRepresentation "#011"
-	tablength=sizeof(TabRepresentation);
 	dbgprintf("Message will now be parsed by fix Snare parser.\n");
 	assert(pMsg != NULL);
 	assert(pMsg->pszRawMsg != NULL);
 
-	/* check if this message is of the type we handle in this (very limited) parser
-
-	 find out if we have a space separated or tab separated for the first item
-	 if tab separated see if the second word is one of our expected tags
-	  if so replace the tabs with spaces so that hostname and syslog tag are going to be parsed properly
-	   optionally replace the hostname at the beginning of the message with one from later in the message
-	  else, wrong message, abort
-	 else, assume that we have a valid timestamp, move over to the syslog tag
-	  if that is tab separated from the rest of the message and one of our expected tags
-	   if so, replace the tab with a space so that it will be parsed properly
-	    optionally replace the hostname at the beginning of the message withone from later in the message 
-
-	*/
+	/* Check if this message is of the type we handle in this parser.
+	 *
+	 * - Find out if the first separator is a tab or a space.
+	 * - If tab-separated, see if the second word is one of our expected tags.
+	 *   - If so, flag as Snare and replace the first tab with space so that hostname and syslog tag parse properly
+	 *   - Else not a snare message.
+	 * - Else assume valid 3164 timestamp, move over to the syslog tag.
+	 * - See if that is tab-separated from one of our expected tags.
+	 *   - If so, flag as Snare.
+	 * - See if either type flagged as Snare.
+	 *   - If so, replace the tab with a space so that it will be parsed properly.
+	 */
+	 
 	snaremessage=0;
-	lenMsg = pMsg->iLenRawMsg - pMsg->offAfterPRI;
 	/* note: offAfterPRI is already the number of PRI chars (do not add one!) */
-	p2parse = pMsg->pszRawMsg + pMsg->offAfterPRI; /* point to start of text, after PRI */
+	lenMsg = pMsg->iLenRawMsg - pMsg->offAfterPRI;
+	/* point to start of text, after PRI */
+	p2parse = pMsg->pszRawMsg + pMsg->offAfterPRI;
 	dbgprintf("pmsnare: msg to look at: [%d]'%s'\n", lenMsg, p2parse);
 	if((unsigned) lenMsg < 30) {
 		/* too short, can not be "our" message */
-		dbgprintf("msg too short!\n");
+		dbgprintf("Message is too short to be Snare!\n");
 		ABORT_FINALIZE(RS_RET_COULD_NOT_PARSE);
 	}
 
-	while(lenMsg && *p2parse != ' ' && *p2parse != '\t' && *p2parse != '#') {
+	/* Check for non-syslog snare format. */
+	while(lenMsg && *p2parse != ' ' && *p2parse != tabRepresentation[0]) {
```

Comment:
since you don't check for both \t and an escaped control character (something starting with #) I don't think this will work if escapecontrolcharacters is enabled